library/derivers.lisp

```diff
@@ -120,3 +120,77 @@
                                                              :name 'coalton:False))))))))
       :location location
       :inline nil))))
+
+(defmethod tc:derive-methods ((class (eql 'classes:hash)) def env)
+  "Deriver implementation for class `Hash'."
+  (let ((location (source:location def)))
+    (list
+     (parser:make-instance-method-definition
+      :name (parser:make-node-variable
+             :location location
+             :name 'classes:hash)
+      :params (list (parser:make-pattern-var
+                     :location location
+                     :name 'x
+                     :orig-name 'x))
+      :body (parser:make-node-body
+             :nodes nil
+             :last-node (parser:make-node-match
+                         :location location
+                         :expr (parser:make-node-variable
+                                :location location
+                                :name 'x)
+                         :branches (mapcar
```

Comment:
this style is inconsistent and difficult to read: MAPCAR over a LOOP-COLLECT is kind of the worst of both worlds in my opinion. Can we just collapse this all into a simple loop?

    (loop :for index :from 0
          :for ctor :in ...
          ...
          :collect ...)